solidity/contracts/Coordinator.sol

```diff
@@ -184,4 +187,51 @@ contract Coordinator {
     require(_amount >= serviceAgreements[_sAId].payment, "Below agreed payment");
     _;
   }
+
+  modifier hasInternalId(uint256 _internalId) {
+    require(callbacks[_internalId].addr != address(0), "Must have a valid internalId");
+    _;
+  }
+
+  function fulfillData(
+    uint256 _internalId,
+    bytes32 _data
+  )
+    public
+    onlyOwner
+    hasInternalId(_internalId)
+    returns (bool)
+  {
+    Callback memory callback = callbacks[_internalId];
+
+    // This is ensuring that ONLY THE FIRST oracle response will be processed
+    // properly, which we are assuming for now, for prototyping. Also, there is
+    // NO HANDLING of LINK transfers or bond recalculations, ATM. TODO: Leaving
+    // that to another story, i.e.
+    // https://www.pivotaltracker.com/story/show/159584617
+    delete callbacks[_internalId];
+
+    // All updates to the oracle's fulfillment should come before calling the
+    // callback(addr+functionId) as it is untrusted. See:
+    // https://solidity.readthedocs.io/en/develop/security-considerations.html#use-the-checks-effects-interactions-pattern
+    return callback.addr.call(callback.functionId, callback.externalId, _data); // solium-disable-line security/no-low-level-calls
+  }
+
+  event CancelRequest(
+    uint256 internalId
+  );
+
+  function cancel(bytes32 _externalId)
```

Comment:
I think we should handle cancelling requests on the Coordinator contract in a separate story after further discussion.

Reply:
OK, removed. https://www.pivotaltracker.com/story/show/161867588